web/src/components/filings-calendar/FilingsCalendarSingleGrid.tsx

```diff
@@ -41,10 +43,51 @@ export const FilingsCalendarSingleGrid = (props: Props): ReactElement => {
     );
   });
 
-  const firstTwoFilings = thisMonthFilings.slice(0, NUM_OF_FILINGS_ALWAYS_VIEWABLE);
+  const visibleFilings = thisMonthFilings.slice(0, NUM_OF_FILINGS_ALWAYS_VIEWABLE);
   const remainingFilings = thisMonthFilings.slice(NUM_OF_FILINGS_ALWAYS_VIEWABLE);
   const isOnCurrentYear = getCurrentDate().year().toString() === props.activeYear;
 
+  const licenseEvent = getLicenseCalendarEvent(
+    parseDate(props.userData?.licenseData?.expirationISO).add(4, "year").toISOString(),
+    date,
+    props.userData.profileData.industryId
+  );
+
+  if (licenseEvent && visibleFilings.length === NUM_OF_FILINGS_ALWAYS_VIEWABLE) {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    remainingFilings.unshift(visibleFilings.pop()!);
+  }
+
+  const renderLicense = (licenseEvent: LicenseCalendarEvent | undefined): ReactNode => {
+    const titles: Record<LicenseEventType, string> = {
+      expiration: Config.licenseDefaults.expirationTitleLabel,
+      renewal: Config.licenseDefaults.renewalTitleLabel,
+    };
+    
+    return licenseEvent ? (
+      <div className="line-height-1 margin-bottom-1" data-testid={`license-${licenseEvent.type}`}>
+        <Tag backgroundColor="accent-warm-extra-light" isHover isRadiusMd isWrappingText>
+          <Link href={`licenses/${licenseEvent.licenseType}-${licenseEvent.type}`}>
+            <a
+              href={`licenses/${licenseEvent.licenseType}-${licenseEvent.type}`}
+              data-testid={licenseEvent.licenseType.toLowerCase()}
+              className="usa-link text-secondary-darker hover:text-secondary-darker text-no-underline"
+            >
+              <span className="text-bold text-uppercase text-base-dark">
+                {Config.dashboardDefaults.calendarFilingDueDateLabel} {licenseEvent.date}
+              </span>{" "}
+              <span className="text-no-uppercase text-underline text-base-dark">
+                {licenseEvent.licenseType} {titles[licenseEvent.type]}
+              </span>
+            </a>
+          </Link>
+        </Tag>
+      </div>
+    ) : (
+      <></>
+    );
```

Comment:
This seems ripe to be its own component.

Also, we avoid `xx ? <stuff> : <></>` notation in favor of `xx && <stuff>`